examples/profile.js

```diff
@@ -23,28 +23,71 @@ app.MainController = function($http, $scope) {
 
   $http.get('data/profile.json').then(angular.bind(this, function(resp) {
     this['profileData'] = resp.data['profile'];
+    this['profilePoisData'] = [
+      {sort: 1, dist: 1000, title: 'First POI', id: 12345},
+      {sort: 2, dist: 3000, title: 'Second POI', id: 12346}
+    ];
   }));
 
+
   /**
-   * @param {Object} item
-   * @return {number}
+   * Factory for creating simple getter functions for extractors.
+   * If the value is in a child property, the opt_childKey must be defined.
+   * The type parameter is used by closure to type the returned function.
+   * @param {T} type An object of the expected result type.
+   * @param {string} key Key used for retrieving the value.
+   * @param {string=} opt_childKey Key of a child object.
+   * @template T
+   * @return {function(Object): T}
    */
-  var z = function(item) {
-    return item['values']['mnt'];
+  var typedFunctionsFactory = function(type, key, opt_childKey) {
```

Comment:
In Angular inner functions are defined at the end of the outer function, using a function declaration rather than a function expression. In your case it would look like this:

```js
app.MainController = function() {
  
  // …

  var elevationExtractor = {
    z: typedFunctionsFactory(types.number, 'mnt', 'values'),
    dist: typedFunctionsFactory(types.number, 'dist')
  };

  // …
  this['point'] = null;

  /**
   * …
   */
  function typedFunctionsFactory(type, key, opt_childKey) {
    // …
  }

};
```

Just a suggestion…